### DISCUSSION TITLE
Compiler for gbnf format?

### DISCUSSION THREAD

**Discussion by cdaman123** [ggml-org/llama.cpp#3666] [General]
Hello,

I am facing some conditions where llama output completes at `:` token. I added markdown in JSON grammar. I want to check state machine of my new grammar regex. Is there any compiler available for `gbnf` like `bnf`. Or what is different b/w the syntax of `bnf` and `gbnf`.

Please help me with resolving this issue.
Thank You.

**Comment by ejones** [comment]
Hey @cdaman123!

For validating the compiled grammar, the output of `main` when used with `--grammar` will contain a debug message with the compiled grammar representation that might be helpful:
```
main: grammar:
root ::= object 
object ::= [{] ws object_11 [}] 
value ::= object | array | string | number | boolean 
array ::= [[] ws array_15 []] 
...
```
This form of the grammar is pretty low level and pretty directly maps to the in-memory representation. 

For something like a specification of the syntax, check out [the GBNF guide at grammars/README.md](https://github.com/ggerganov/llama.cpp/blob/master/grammars/README.md) if you haven't already.

Let  me know if that helps or if you were after something else.
